### PR TITLE
Add plugin for bottle

### PR DESCRIPTION
**Pull request by jiang1997**
- [x] Add a test case for the new plugin
- [x] Add a CHANGELOG entry for the new plugin
- [x] Add a component id in [the main repo](https://github.com/apache/skywalking/blob/master/oap-server/server-bootstrap/src/main/resources/component-libraries.yml#L415)
- [x] Add a logo in [the UI repo](https://github.com/apache/skywalking-rocketbot-ui/tree/master/src/views/components/topology/assets)
- [x] Rebuild the `requirements.txt` by running `tools/env/build_requirements_(linux|windows).sh`
- [x] Rebuild the `Plugins.md` documentation by running `make doc-gen`

### CODE REVIEW

**Comment by Superskyyy** [comment]
Thanks for the contribution, please fix the lint and update your base branch.

**Comment by jiang1997** [comment]
my fault 😂

**Comment by Superskyyy** [comment]
> my fault 😂
Not a problem！
Please do a reformat if you are using pycharm for files **you added**, it will fix most things.

**Comment by Superskyyy** [comment]
Please also add a entry in the Changelog thanks

File: skywalking/plugins/sw_bottle.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from skywalking import Layer, Component, config
+from skywalking.trace.carrier import Carrier
+from skywalking.trace.context import get_context, NoopContext
+from skywalking.trace.span import NoopSpan
+from skywalking.trace.tags import TagHttpMethod, TagHttpURL
+
+link_vector = ['http://bottlepy.org/docs/dev/']
+support_matrix = {
+    'bottle': {
+        '>=0.12': ['*']
```
Comment (Superskyyy) [inline skywalking/plugins/sw_bottle.py:27]: This isn't a supported syntax, and you confused the left and right side. 

left side should be version of Python right side is the package, see and example below.
```
support_matrix = {
    'hug': {
        '>=3.10': ['2.5', '2.6'],  # api deprecated for 3.10
        '>=3.6': ['2.4.1', '2.5', '2.6'],  # support begins 2.4.1
    }
}
```

File: docs/en/setup/Plugins.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ or a limitation of SkyWalking auto-instrumentation (welcome to contribute!)
 Library | Python Version - Lib Version | Plugin Name
 | :--- | :--- | :--- |
 | [aiohttp](https://docs.aiohttp.org) | Python >=3.10 - NOT SUPPORTED YET; Python >=3.6 - ['3.7.4'];  | `sw_aiohttp` |
+| [bottle](http://bottlepy.org/docs/dev/) | Python >=0.12 - ['*'];  | `sw_bottle` |
```
Comment (Superskyyy) [inline docs/en/setup/Plugins.md:17]: see here you can tell the generated doc isn't correct.

Comment (jiang1997) [reply docs/en/setup/Plugins.md:17]: got it.

Comment (jiang1997) [reply docs/en/setup/Plugins.md:17]: Is there any other error reported in Actions? I want to eliminate as many mistakes as possible on each commit. Thanks!

Comment (wu-sheng) [reply docs/en/setup/Plugins.md:17]: It is better you could run the tests locally. Then you could catch all errors.

Comment (jiang1997) [reply docs/en/setup/Plugins.md:17]: > It is better you could run the tests locally. Then you could catch all errors.

Yea, I make it now.
Here is the problem I encountered.
```
docker pull apache/skywalking-python-agent:latest-plugin
```
The above command always runs into trouble with the error message below
```
Error response from daemon: pull access denied for apache/skywalking-python-agent, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```
Seems the latest tag is unavailable, and docker-compose.base.yml under the tests/plugin folder needs to be altered for the test in my situation.

Comment (jiang1997) [reply docs/en/setup/Plugins.md:17]: > > Is there any other error reported in Actions? I want to eliminate as many mistakes as possible on each commit. Thanks!
> 
> not sure, but seems like you are almost there, remember to regenerate docs and add a changelog

bear in mind

**Review by jiang1997 — COMMENTED** [review]
The unit test refers to Django's.

File: docs/en/setup/Plugins.md (hunk 2 above)
Comment (kezhenxu94) [reply docs/en/setup/Plugins.md]: > > It is better you could run the tests locally. Then you could catch all errors.
> 
> Yea, I make it now. Here is the problem I encountered.
> 
> ```
> docker pull apache/skywalking-python-agent:latest-plugin
> ```
> 
> The above command always runs into trouble with the error message below
> 
> ```
> Error response from daemon: pull access denied for apache/skywalking-python-agent, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
> ```
> 
> Seems the latest tag is unavailable, and docker-compose.base.yml under the tests/plugin folder needs to be altered for the test in my situation.

Sorry for my belated response. This Docker image is built on the fly right before running the tests. Please refer to https://github.com/apache/skywalking-python/blob/88b30ae094079167409e7c2bf4ff68431bc9e96a/.github/workflows/CI.yaml#L135

Comment (kezhenxu94) [reply docs/en/setup/Plugins.md]: > Is there any other error reported in Actions? I want to eliminate as many mistakes as possible on each commit. Thanks!

Hi. Can you see the actions log [here](https://github.com/apache/skywalking-python/runs/7165080924?check_suite_focus=true#step:6:177)? There is a diff output that indicates what’s the expected data and what’s the actual data (ignore the matching pattern like `gt 0`, `not null`, etc.)

**Review by jiang1997 — COMMENTED** [review]
Here is the way I create "apache/skywalking-python-agent:latest-plugin" image for the test.
First, create the file as below,
skywalking-python/Dockerfile.test
```
FROM python:3.8

WORKDIR /usr/src/app

COPY . .

RUN pip install -r requirements.txt
RUN pip install -e .[test]
```
then
```
docker build -t apache/skywalking-python-agent:latest-plugin .  -f Dockerfile.test 
```

**Comment by jiang1997** [comment]
> 



> Here is the way I create "apache/skywalking-python-agent:latest-plugin" image for the test. First, create the file as below, skywalking-python/Dockerfile.test
> 
> ```
> FROM python:3.8
> 
> WORKDIR /usr/src/app
> 
> COPY . .
> 
> RUN pip install -r requirements.txt
> RUN pip install -e .[test]
> ```
> 
> then
> 
> ```
> docker build -t apache/skywalking-python-agent:latest-plugin .  -f Dockerfile.test 
> ```

The reason  I mention this is that the instruction to run a test locally is kind of vague by now.

**Comment by kezhenxu94** [comment]
> The reason I mention this is that the instruction to run a test locally is kind of vague by now.

I'm sorry to hear that... and I've created [an issue](https://github.com/apache/skywalking/issues/9301) to track this issue, if you're interested in helping to complete the doc, feel free to open a PR and add the guide about running tests locally.

Thanks for add this plugin anyway!

**Review by kezhenxu94 — APPROVED** [review]
LGTM, @Superskyyy wanna recheck?

**Comment by jiang1997** [comment]
> > The reason I mention this is that the instruction to run a test locally is kind of vague by now.
> 
> I'm sorry to hear that... and I've created [an issue](https://github.com/apache/skywalking/issues/9301) to track this issue, if you're interested in helping to complete the doc, feel free to open a PR and add the guide about running tests locally.
> 
> Thanks for add this plugin anyway!

I'm glad to go through with these.

**Comment by wu-sheng** [comment]
@jiang1997 I think you missed two steps besides this PR.

1. Add component ID into the main repo.
2. Add logo(if it has one) into booster UI. Ping @kezhenxu94 It seems this PR template is still leading to rocketbot UI.

**Comment by jiang1997** [comment]
> @jiang1997 I think you missed two steps besides this PR.
> 
> 1. Add component ID into the main repo.
> 2. Add logo(if it has one) into booster UI. Ping @kezhenxu94 It seems this PR template is still leading to rocketbot UI.



> @jiang1997 I think you missed two steps besides this PR.
> 
> 1. Add component ID into the main repo.
> 2. Add logo(if it has one) into booster UI. Ping @kezhenxu94 It seems this PR template is still leading to rocketbot UI.

So I am supposed to do those before this merge, right?

**Comment by wu-sheng** [comment]
Yes, at least the component ID should get merged first.

**Pull request merged**